lib/src/lock_file.dart

```diff
@@ -244,4 +273,14 @@ class LockFile {
 ${yamlToString(data)}
 """;
   }
+
+  /// Returns the dependency classification for [package].
+  String _dependencyType(String package) {
+    if (_mainDependencies.contains(package)) return 'direct main';
+    if (_devDependencies.contains(package)) return 'direct dev';
+    if (_overriddenDependencies.contains(package)) {
+      return 'direct overridden';
```

Comment:
> Bikeshed: It may be a little easier on tools if these are single-word identifiers. How about "main", "dev", "overridden", and "transitive"?

I think we agreed on this scheme in https://github.com/dart-lang/pub/issues/1721#issuecomment-337074828.

> Also, I'm guessing it's deliberate that main and dev take priority over overridden. That seems right to me too. It would be good to leave a comment to that effect.

Done.